Added a selection property for a special communication pattern

mwelzl opened the pull request:
.. that gets in the way of using streams when cloning.
Closes #441.

philsbln commented on draft-ietf-taps-interface.md:
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
+client actively opening a connection, then sending data to the server. The
+server listens (passive open), reads, and then answers. This property
+specifies whether an application wants to diverge from this pattern -- either
+by actively opening, immediately followed by reading, or passively opening,
+immediately followed by writing.
```
I get the problem for read-after-active-open, but what is the problem with passive open and immediately writing – this is fairly common in TCP based protocols and you can easily put the write in the connection ready.

mwelzl replied:
It's the same thing with multi-streaming: a stream only comes into being when the first message is sent on it. So, "passive open and immediately writing" doesn't map onto anything with streams - there is no such thing as a stream becoming available without data arriving on it.

philsbln commented on draft-ietf-taps-interface.md:
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
 on receiving them.
 
 
+### Active side is not the first to write {#active-read-before-send}
```
```suggestion
### Initiating side is not the first to write {#active-read-before-send}
```

csperkins commented:
How does this interact with `Rendezvous()`?

csperkins commented on draft-ietf-taps-interface.md:
```diff
@@ -1124,6 +1125,30 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
+
+Default:
+: Ignore
+
```
I wonder if this should be `Type: Enumeration`, with values `InitiatorSendsFirst` (the default) and `ListenerSendsFirst`?

mwelzl replied:
Why make it so limiting? Preference allows for values such as Prefer, Avoid, Require, ...

csperkins replied:
How does a preference work here? If the application protocol requires the listener to send first, then it need to do this or fail.

mwelzl commented:
@csperkins - this now states that it is irrelevant for Rendezvous(), and explicitly mentions the Initiate() and Listen() calls when talking about actively and passively opening a Connection.

Why is it irrelevant?

Remember, the problem is that you actively open a Connection that could be mapped to a stream, in which case an active Open doesn't do anything on the wire - and then, right away, you want to read. Then the peer never heard from you, you haven't even tried sending anything to it.

Enabling this property will allow the TAPS system to carry out such a Connection = stream mapping. When calling rendezvous(), however, just using a stream is out of the question - you need signaling, and chances are that you're going to actively send stuff in any case. In other words, I don't think there's a mapping that would work with rendezvous() that enabling this property would allow, and hence the property is irrelevant for rendezvous().

mwelzl commented:
Merging this now: I think I have addressed the comment by @csperkins. We can always fix things later again if we must - but for now, this closes #441, and it also relates to #508, which gates #507.... so it's probably better to move forward at this point.

csperkins commented on draft-ietf-taps-interface.md:
```diff
@@ -1124,6 +1125,31 @@ not all ICMP errors will necessarily be delivered, so applications cannot rely
+specifies whether an application wants to diverge from this pattern -- either
+by actively opening with Initiate(), immediately followed by reading, or passively opening with Listen(),
+immediately followed by writing. This property is irrelevant for connection
+establishment using Rendezvous().
```
"irrelevant" isn't right here - maybe "is ignored"?

mwelzl replied:
ACK, fixed with a commit to the master
